config/v1/types_ingress.go

```diff
@@ -131,6 +131,7 @@ type LoadBalancer struct {
 
 // AWSIngressSpec holds the desired state of the Ingress for Amazon Web Services infrastructure provider.
 // This only includes fields that can be modified in the cluster.
+// +openshift:validation:FeatureGateAwareXValidation:featureGate=InstallEIPForDefaultNLBIngressController,rule="self.type == 'NLB' ? true : !has(self.networkLoadBalancerParameters)",message="networkLoadBalancerParameters are forbidden when type is not NLB."
```

Comment:
Cross link to https://github.com/openshift/enhancements/pull/1634#discussion_r1814173039. I think this CEL is okay for now (and better to keep it here), but I might have to remove it later depending on the designs of the subnets API.